test/e2e/test/apmserver/checks_k8s.go

```diff
@@ -123,22 +123,76 @@ func CheckServicesEndpoints(b Builder, k *test.K8sClient) test.Step {
 func CheckSecrets(b Builder, k *test.K8sClient) test.Step {
 	return test.CheckSecretsContent(k, b.ApmServer.Namespace, func() []test.ExpectedSecret {
 		apmName := b.ApmServer.Name
+		apmNamespace := b.ApmServer.Namespace
 		// hardcode all secret names and keys to catch any breaking change
 		expected := []test.ExpectedSecret{
-			{Name: apmName + "-apm-config", Keys: []string{"apm-server.yml"}},
-			{Name: apmName + "-apm-token", Keys: []string{"secret-token"}},
+			{
+				Name: apmName + "-apm-config",
+				Keys: []string{"apm-server.yml"},
+				Labels: map[string]string{
+					"apm.k8s.elastic.co/name":    apmName,
+					"common.k8s.elastic.co/type": "apm-server",
+				},
+			},
+			{
+				Name: apmName + "-apm-token",
+				Keys: []string{"secret-token"},
+				Labels: map[string]string{
+					"apm.k8s.elastic.co/name":        apmName,
+					"common.k8s.elastic.co/type":     "apm-server",
+					"eck.k8s.elastic.co/credentials": "true",
+				},
+			},
 		}
 		if b.ApmServer.Spec.ElasticsearchRef.Name != "" {
 			expected = append(expected,
-				test.ExpectedSecret{Name: apmName + "-apm-es-ca", Keys: []string{"ca.crt", "tls.crt"}},
-				test.ExpectedSecret{Name: apmName + "-apm-user", Keys: []string{b.ApmServer.Namespace + "-" + apmName + "-apm-user"}},
+				test.ExpectedSecret{
+					Name: apmName + "-apm-es-ca",
+					Keys: []string{"ca.crt", "tls.crt"},
+					Labels: map[string]string{
+						"apmassociation.k8s.elastic.co/name":        apmName,
+						"apmassociation.k8s.elastic.co/namespace":   apmNamespace,
```

Comment:
CA copy also contains `apmassociation.k8s.elastic.co/type: elasticsearch` , not sure if you removed it on purpose ?

Reply:
I don't see that label on the resource. Is it something we used to have?

Reply:
Ah, is it part of https://github.com/elastic/cloud-on-k8s/pull/3043?

Reply:
Sorry, I forgot to add some context here. Let's take this one into consideration later when #3043 is merged.